appeals/web/src/server/appeals/appeal-details/address/__tests__/address.test.js

```diff
@@ -0,0 +1,215 @@
+import { parseHtml } from '@pins/platform';
+import supertest from 'supertest';
+import { appealData } from '#testing/app/fixtures/referencedata.js';
+import { createTestEnvironment } from '#testing/index.js';
+import nock from 'nock';
+
+const { app, installMockApi, teardown } = createTestEnvironment();
+const request = supertest(app);
+const baseUrl = '/appeals-service/appeal-details';
+
+describe('neighbouring-sites', () => {
+	beforeEach(installMockApi);
+	afterEach(teardown);
+
+	describe('GET /add', () => {
+		it('should render updateSiteAddress page', async () => {
+			const appealId = appealData.appealId.toString();
+			const response = await request.get(
+				`${baseUrl}/${appealId}/appellant-case/site-address/change/1`
+			);
+			const element = parseHtml(response.text);
+
+			expect(element.innerHTML).toMatchSnapshot();
+		});
+	});
+
+	describe('POST /change/:siteId', () => {
+		it('should re-render updateSiteAddress page if addressLine1 is null', async () => {
+			const appealId = appealData.appealId.toString();
+
+			const invalidData = {
+				addressLine1: null,
+				addressLine2: null,
+				town: 'London',
+				county: null,
+				postCode: 'E1 8RU'
+			};
+			const response = await request
+				.post(`${baseUrl}/${appealId}/appellant-case/site-address/change/1`)
+				.send(invalidData);
+
+			expect(response.statusCode).toBe(200);
+
+			const element = parseHtml(response.text);
+
+			expect(element.innerHTML).toMatchSnapshot();
+			expect(element.innerHTML).toContain('Enter the first line of the address');
```

Comment:
this is perfectly good, just a suggestion that you could use the `govuk-error-summary` class as a selector to get just that element and then check for the text inside that